lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -495,12 +512,17 @@ def __init__(self, model: str, model_uuid: str, application: str, charm_name: st
             model: a string name of the Juju model
             model_uuid: a globally unique string identifier for the Juju model
             application: an application name as a string
+            unit: a unit name as a string
             charm_name: name of charm as a string
         """
         self.model = model
         self.model_uuid = model_uuid
         self.application = application
         self.charm_name = charm_name
+        self.unit = unit
+
+        if not self.unit and not self.charm_name:
```

Comment:
1. No, we are not, though after discussion with @mmanciop, it's probably ok to have both.

2. Realistically, this is because, without declaring an explicit `Type`, Python doesn't have an amazing way to represent "one or the other of these is required", and we won't know what the label should be if it's a bare string. The exception is thrown since `JujuTopology` isn't useful without at least one of them being set, and the "other" Pythonic way other than throwing an exception is to have a completely different `@classmethod` constructor

Reply:
My main objection here is that we are marking two constructor arguments as optional but raising an exception when one of them is missing.

Reply:
Now, only `application` is required, to support workload-only charms. Either `charm`, `unit`, both, or neither are otherwise acceptable.